### ISSUE TITLE
Mock up an agreement creation form for budget approver story

### ISSUE THREAD

**Issue by cmajel** [18F/NRM-Grants-Agreements#173]
**Background**
In #169, we identified that in our MVP, we want a budget approver to be able to approve a commitment within the system.  #184 captures the user needs and technical de-risking focus areas for proposal creation. 

In an electronic agreement creation world, what data needs to be recorded ahead of time so that a budget approver has the information they need to review and approve a commitment? This issue focuses on creating a mock of what that agreement creation form needs to include, and will build off #153, which focuses on showing minimum proposal fields.

**We'll know we're done when...**
- [ ] We have a mock that we can test...
   - with program managers,  to validate if they could fill out the form needed to create an agreement and send budget needed info
   - with budget approvers to validate if this would give them the information they need

**Tasks**
- [x] Identify if additional fields are needed from #153, or if those fields are sufficient. 
- [x] Identify outgoing fund agreements that are most common 
- [x] Work with Jay to choose an outgoing agreement type  / specific use case to focus in on for this sprint
- [x] Reconcile with #162 , how does this approach support, conflict, or adjust our as-is workflow model?
- [x] Low fidelity mock or field list exists and shared with Jay/Jamie for feedback
- [x] A mock exists and can be shared with users

**Open questions**
- [X] Should this include the work of PM entering commitment data? **YES!**
- [X] What agreement type should we focus on? 
- [ ] Should this include mocks of what a budget approver would see when they log in? Or just mocks of the form? 
- [ ] Do we want to add an issue for a to-be workflow model in parallel?

**Comment by MelissaBraxton** [comment]
Program Managers who create participating agreements will be mostly on the NFS side. The work you'd do under one of these is typically on NFS land.

**Comment by allisonnorman** [comment]
Since we added the Program Manager user story, I added that here as it reflects the creation process needs more than the Budget Approver story we initially started with.

**Comment by cmajel** [comment]
Walking through with Jay today (4/21). [Mocks are available in Figma](https://www.figma.com/file/AOWq8IWGSAUGZiqZ1nZSeF/USFS-Grants-and-Agreements-Playground?node-id=325%3A0) (follow the pink line) and are based on the `happy path` work in #162. 

Some assumptions we're making in these mocks:
- The person filling out the form is a program approver on a desktop / non mobile device
- They have a workplan ID for the project, and know it's a participating agreement
- This form would allow the program approver to create the proposal and approve the commitment.

Hypotheses we want to validate:
- That when a program approver is creating a proposal, we can combine multiple steps (program acceptance, program approval, program commitment approval) while maintaining proper checks and balances.
- That prepopulating data from Workplan from the start is valuable to people creating new proposals.
- That breaking the form into multiple thematic pages with conditional logic will help program managers know what's required of them. 

And, lots of questions in Figma, but some of the bigger ones:
- Do we have the right list of agreement types? How often is this known? What path do we want when it is not known?
- Can we determine the agreement team (program manager, budget approver, grants specialist) at proposal creation? What checks are needed? 
- What data needs to be consistent between Workplan and G&A? For example, are there commitment fields like BOC, Budget org code, and job code that should depend on which Workplan is linked?
- How should we handle optional fields? Do we include them when creating a proposal, or offer them up later once a proposal / agreement becomes more developed?
  -  Related: which fields are optional for a PA, and which are not valuable at all? For example, could a PA be used for research, and therefore we'd need to screen for that?

**Comment by cmajel** [comment]
We reviewed this again with G&A staff on 4/23, and got some good feedback. [Mocks are available in Figma](https://www.figma.com/file/AOWq8IWGSAUGZiqZ1nZSeF/USFS-Grants-and-Agreements-Playground?node-id=325%3A0) (follow the pink line) | [Debrief notes](https://docs.google.com/document/d/1U-Auozuf2XSIHHdi6NPK7X7FjqIxNeEe6SuDq4zamEI/edit#heading=h.vr6xiyb7ldku) 

In addition to specific UI adjustments, here are some key things we learned and questions to consider in the next round of iteration:
* In our updated iterator, we hypothesized that we could route workplans to individuals so that they can create proposals. But, Workplan's POC field is not consistent, and org code may be a very large list. 
  * How do we get people to the right Workplans for proposal creation?
* We assumed that the proposal creator knew it was a participating agreement, but we learned that agreement type is not always known, and we may want to consider ways to help learn more basic facts about a proposal to guide folks towards the right agreement type, and provide useful context to GMSs.
  * What do we need to ask to guide folks to the right agreement type?
  * How do we loop in GMSs earlier in the process so that they can collaborate sooner and confirm agreement details earlier to catch errors?
* There is value to getting an agreement number earlier, and some of the ways folks work around the system are to get to an agreement number sooner. Agreement number is how work gets tracked and more familiar than a proposal id.
  * How can we get to an agreement number sooner?
* Our `agreement team` page provides the opportunity to assign an individual, but some regions assign approvers based on `who's next in the queue`.
  * How do we support regional / forest differences in how they assign grants staff?

cc @MelissaBraxton - anything to add? I think we can close this out and create a new issue for refining the mock in the next sprint.
